### PR TITLE
Check PodIP against PodCIDR only when hostNetwork=false.

### PR DESCRIPTION
**Pull request by sawsa307**
/assign @swetharepakula
/assign @gauravkghildiyal

### CODE REVIEW

**Review by swetharepakula — COMMENTED** [review]
can you add tests for this?

**Comment by gauravkghildiyal** [comment]
Thanks

/lgtm

**Comment by sawsa307** [comment]
Just added test cases to ensure we don't filter out pods if it uses hostNetwork. Please take a look!

File: pkg/neg/syncers/utils.go [hunk 1]
```diff
@@ -568,7 +568,7 @@ func validatePod(pod *apiv1.Pod, nodeLister, serviceLister cache.Indexer, networ
 		count[negtypes.OtherError]++
 		return count, negtypes.ErrEPNodeTypeAssertionFailed
 	}
-	if err = nodeContainsPodIP(node, networkEndpoint); err != nil {
+	if err = nodeContainsPodIP(node, networkEndpoint); !pod.Spec.HostNetwork && err != nil {
```
Comment (aojea) [inline pkg/neg/syncers/utils.go:571]: I don't know how the golang compiler translate this, but is also easier to understand if you only execute `nodeContainsPodIP(node, networkEndpoint)` if `!pod.Spec.HostNetwork`

```suggestion
        if  !pod.Spec.HostNetwork {
            if err = nodeContainsPodIP(node, networkEndpoint); && err != nil {
        }
```

Comment (gauravkghildiyal) [reply pkg/neg/syncers/utils.go:571]: Seems + reads better

Comment (sawsa307) [reply pkg/neg/syncers/utils.go]: Yes I agree. This makes the check easier to read. Updated.

**Review by swetharepakula — COMMENTED** [review]
Thanks!

/approve

**Pull request merged**